### ISSUE TITLE
[Accessibility] An element of the given ControlType must support the Toggle pattern for PropertyGrid

### ISSUE THREAD

**Issue by Amy-Li03** [dotnet/winforms#4428]
* .NET Core Version: 
.Net SDK 6.0.100-alpha.1.20623.10

* Have you experienced this same bug with .NET Framework?: 
No

* More info:
1. This issue doesn't repro on .Net core 3.1 & .Net 5.0
2. In Inspect, the IsTogglePattermAvailable property is **false** both .Net framework and .Net core projet.

**Minimal repro:**
[DemoApp.zip](https://github.com/dotnet/winforms/files/5764054/DemoApp.zip)

1. Create a winforms .NET core project with targeting 6.0
2. Add a PropertyGrid control to the form designer
3. Build and run the core project
4. Observe PropertyGrid using the Accessibility Insight

**Actual behavior:**
AccessibilityInsights detected an error: An element of the given ControlType must support the Toggle pattern
![Core](https://user-images.githubusercontent.com/62929087/103522352-37f85200-4eb5-11eb-8f29-93c641f5e90f.PNG)

**Expected behavior:**
AccessibilityInsights should not detect this error.

**Comment by merriemcgaw** [comment]
These are toggle buttons and as such should have the Toggle Pattern implemented. This seems like a regression from previous behavior.

@Amy-Li03 can you clarify - what is the control type and is the TogglePattern available for these particular buttons in 4.8, Core 3.1, Core 5 and 6? Is this just a recent .NET 6 regression?

**Comment by Amy-Li03** [comment]
@merriemcgaw 
These particular buttons' control type is **button** and IsTogglePatternAvailable is **false** in .Net framework 4.8, .Net core 3.1 & .Net 5.0. But in latest .Net 6.0, control type is **check box** and IsTogglePatternAvailable is **false**

Yes, it is a recent .Net 6.0 regression, in previous build: 6.0.100-alpha.1.20567.24, these buttons' control type is button and there is no error when detecting by Accessibility Insight.

**Comment by SergeySmirnov-Akvelon** [comment]
Hi @merriemcgaw. During investigation this bug, I found that in 4.7.2 these buttons were also of the `Checkbox` control type, which changed to `Button` from 4.8. After fixing #3618 we returned the behavior that was originally in 4.7.2. We can return the  `Button` control type, or leave the `Checkbox` control type and implement the `Toggle` pattern. What is the expected behavior?

**Comment by merriemcgaw** [comment]
@SergeySmirnov-Akvelon - do you know if we can add the Toggle pattern to the button control? Is there any reason to think we could not/should not?

**Comment by SergeySmirnov-Akvelon** [comment]
Hi @merriemcgaw. Yes, we can do it and I implemented it in the #4586. Could you please take a look at #4586 and confirm that this is the expected behavior?

**Comment by merriemcgaw** [comment]
@SergeySmirnov-Akvelon, that looks great in my eyes! Thanks so much! If a developer doesn't want to set it as a toggle button, they can get the regular behavior, right?

**Comment by SergeySmirnov-Akvelon** [comment]
Hi @merriemcgaw. It seems to me that there is no easy way for users to influence how these buttons work.

On the one hand, we have the old behavior in the framework, where the buttons are of the `checkbox` control type and our change may confuse the user.
On the other hand, these controls really look more like buttons than checkboxes. In addition, they have the behavior of radio buttons, when you check one button, the other button loses the `checked` state. Therefore, the `button` control type and the `toggle` pattern look more logical than the `checkbox` control type.

What do you think about it?

**Comment by merriemcgaw** [comment]
I absolutely agree. That seems like the best approach in my book. Thanks @SergeySmirnov-Akvelon !

**Comment by SergeySmirnov-Akvelon** [comment]
Hi @merriemcgaw. The CTI team found an unpleasant surprise. According to the Accessibility Insights rules, the button cannot support `Invoke` and `Toggle` patterns at the same time, as a result we see an error: 
![Issue-4428-AIError](https://user-images.githubusercontent.com/23376742/109616293-0e227b00-7b46-11eb-9966-79fc7fd11a70.png)
Probably we will have to go back to using the `Checkbox` control type with support for the `Toggle` pattern. Are you satisfied with this behavior?

**Comment by SergeySmirnov-Akvelon** [comment]
Hi @merriemcgaw. I've updated the behavior of the buttons to be of the `RadioButton` control type. `Select` and `AddToSelection` methods now have the old `Toggle` method behavior. `RemoveFromSelection` does nothing, as we should always have one option selected.
![Issue-4428-RadioButton](https://user-images.githubusercontent.com/23376742/110473665-d7bda080-80ef-11eb-8d8d-deafb5449ce5.gif)
![Issue-4428-radiobutton](https://user-images.githubusercontent.com/23376742/110473682-dc825480-80ef-11eb-9dd6-e7f9fc94ac69.png)
![Issue-4428-radiobutton-AI](https://user-images.githubusercontent.com/23376742/110473700-e1470880-80ef-11eb-949e-4569edc73eb3.png)

**Comment by Zheng-Li01** [comment]
Verified the issue with latest 6.0.100-preview.3.21169.6, the issue has been fixed that the ControlType has been changed as Radio button, and everything working as expected.
![image](https://user-images.githubusercontent.com/38325459/111955226-350c1580-8b24-11eb-8a26-2f1950db2e3e.png)
